sale_order_create_event_session_number/models/sale_order.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+# (c) 2017 Alfredo de la Fuente - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from openerp import models, api, fields, exceptions, _
+from openerp.addons.event_track_assistant._common import _convert_to_utc_date
+from dateutil.relativedelta import relativedelta
+
+
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+
+    @api.multi
+    def action_button_confirm(self):
+        for sale in self.filtered('project_id'):
+            sale_lines = sale.order_line.filtered(
+                lambda x: x.recurring_service)
+            for line in sale_lines:
+                if not line.end_date and not line.session_number:
+                    raise exceptions.Warning(
+                        _("You must enter the number of sessions, or the end"
+                          " date, for line: '%s'") % (line.name))
+            sale_lines = sale.order_line.filtered(
+                lambda x: x.recurring_service and x.session_number > 0)
+            self._generate_lines_end_date(sale_lines)
+        return super(SaleOrder, self).action_button_confirm()
+
+    def _generate_lines_end_date(self, lines):
+        task_obj = self.env['project.task']
+        tz = self.env.user.tz
+        for line in lines:
+            if line.project_by_task == 'yes':
```

Comment:
`fec_ini = line.project_by_task == 'yes' and _convert_to_utc_date(line.start_date, line.start_hour, tz=tz) or fields.Datetime.from_string(self.project_id.date_start)`